iceoryx_examples/request_response/README.md

```diff
@@ -0,0 +1,228 @@
+
+### Client using Waitset
+
+At first, the includes for the client port, request-response types, WaitSet, and runtime are needed.
+<!-- [geoffrey] [iceoryx_examples/request_response/client_cxx_waitset.cpp] [iceoryx includes] -->
+```cpp
+#include "request_and_response_types.hpp"
+
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_posh/popo/client.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+```
+
+Next, the iceoryx runtime is initialized. With this call,
+the application will be registered at `RouDi`, the routing and discovery daemon.
+<!-- [geoffrey] [iceoryx_examples/request_response/client_cxx_waitset.cpp] [initialize runtime] -->
+```cpp
+iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+```
+
+After creating the runtime, the client port is created and attached to the Waitset.
+The [options](https://iceoryx.io/latest/getting-started/examples/iceoptions/) can be used to alter the behavior of the client, like setting the response
+queue capacity or blocking behavior when the response queue is full or the server is too slow.
+The `ClientOptions` are similar to `PublisherOptions`/`SubscriberOptions`.
+
+<!--[geoffrey][iceoryx_examples/request_response/client_cxx_waitset.cpp][create waitset]-->
+```cpp
+iox::popo::WaitSet<> waitset;
+
+iox::popo::ClientOptions options;
+options.responseQueueCapacity = 2U;
+iox::popo::Client<AddRequest, AddResponse> client({"Example", "Request-Response", "Add"}, options);
+
+// attach client to waitset
+waitset.attachState(client, iox::popo::ClientState::HAS_RESPONSE).or_else([](auto) {
+    std::cerr << "failed to attach client" << std::endl;
+    std::exit(EXIT_FAILURE);
+});
+```
+
+The main goal of the client is to request from the server the sum of two numbers that the
+client sends. When the sum is received from the server, the received sum is re-used to insert
+it to the `addend` of the next request to send.
+This calculates a Fibonacci sequence.
+First a request is prepared and send to the server. The client use then the `Waitset`
+to wait for a response from the server.
+Once a response is received, the client iterates through the notification vector to collect
+the samples and process the response.
+<!-- [geoffrey] [iceoryx_examples/request_response/client_cxx_waitset.cpp] [[mainloop]] -->
+```cpp
+while (!iox::posix::hasTerminationRequested())
+{
+    // send request
+    // We block and wait for samples to arrive, when the time is up we send the request again
+    auto notificationVector = waitset.timedWait(iox::units::Duration::fromSeconds(5));
+
+    for (auto& notification : notificationVector)
+    {
+        if (notification->doesOriginateFrom(&client))
+        {
+        // receive response
+        }
+    }
+    constexpr std::chrono::milliseconds SLEEP_TIME{950U};
+    std::this_thread::sleep_for(SLEEP_TIME);
+}
+```
```

Comment:
I would remove this part here and add the loop over the notification vector to the `take response` snippet. In the corresponding paragraph you could then explain that 
- we wait for samples to arrive via `timedWait`
- we iterate over the notification vector to check if we were triggered from our client
- and if so, take the response... (already described in 115 ff)